components/Carousel/Carousel.js

```diff
@@ -1,8 +1,65 @@
 class Carousel {
+  constructor(element) {
+    this.element = element;
+    this.images = this.element.getElementsByTagName('img');
+
+    // convert this.images into a real array
+    this.images = Array.from(this.images);
+
+    this.currentIndex = 0;
+    this.images[this.currentIndex].style.display = 'block';
+
+    this.leftButton = this.element.querySelector('.left-button');
+    this.rightButton = this.element.querySelector('.right-button');
+    this.leftButton.addEventListener('click', (event => {
+      this.leftSelected(event);
+    }));
+    this.rightButton.addEventListener('click', (event => {
+      this.rightSelected(event);
+    }));
+  }
+
+
+  leftSelected() {
+    // set display on all images to null
+    this.images.map(image => image.style.display = 'none');
+
+    this.currentIndex -= 1;
+    if (this.currentIndex < 0) {
+      this.currentIndex = this.images.length - 1;
+    }
+    console.log(this.currentIndex)
+    //set current image to display
+    this.images[this.currentIndex].style.display = 'block';
+  }
+
+  rightSelected() {
+    // set display on all images to null
+    this.images.map(image => image.style.display = 'none');
+
+    this.currentIndex += 1;
+    if (this.currentIndex > this.images.length - 1) {
+      this.currentIndex = 0;
+    }
+
+    //set current image to display
+    this.images[this.currentIndex].style.display = 'block';
+  }
+
 
 }
 
-let carousel = document.querySelector();
+let carousels = document.querySelectorAll('.carousel');
+carousels = Array.from(carousels).map(carousel => new Carousel(carousel));
+
```

Comment:
Amazing work on the carousel here! The comments you added are incredibly helpful as well 👍